### PR TITLE
docs: Update Autocomplete in DG

### PR DESCRIPTION
**Pull request by justgnohUG**
Notable changes:
- Refactor-ed Autocomplete DG
- Added images to illustrate expected Ui behaviors
- Added Future Enhancements

For partial completion of #90

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -230,22 +230,49 @@ layoutBounds calculations.
 
 ### Autocomplete
 
-The current implementation consists of an AutoCompleteListPanel which is made up of AutoCompleteCells.
-Each AutoCompleteCell contains a command word. Command words are retrieved by LogicManager and populated
-by MainWindow. Event filters are added to the root by MainWindow and the corresponding keys (`Tab`, `UP`, `DOWN`)
+The current implementation consists of a Ui componenet called `AutoCompleteListPanel` which is made up of `AutocompleteCells`.
```
Comment (oeiyiping) [inline docs/DeveloperGuide.md:233]: Slight typo for component

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -230,22 +230,49 @@ layoutBounds calculations.
...
+Each `AutoCompleteCell` contains a command word. Command words are retrieved by calling `getAutocompleteCommands()` in `LogicManager` and populated
+by `MainWindow` in the `fillInnerParts()` method. 
+
+In `CommandBox` a method called `setKeyUpCallback()` triggers the method `updateList()` in `AutocompleteListPanel` on every release of a key. This updates the existing command panel
+with the correct filtered commands.
+- The `setKeyUpCallback()` uses `addEventFilter()` and detects when a key is released before triggering the function to handle it.
+
+Event filters are added to the root by `MainWindow` and the corresponding keys (`Tab`, `UP`, `DOWN`)
```
Comment (oeiyiping) [inline docs/DeveloperGuide.md:241]: Maybe capitalize `Tab`  for consistency with the other two keys?

**Review by oeiyiping — COMMENTED** [review]
Aside from the minor points mentioned in the comments, LGTM.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (justgnohUG) [reply docs/DeveloperGuide.md:233]: Fixed! Thanks

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (justgnohUG) [reply docs/DeveloperGuide.md:241]: Fixed!

**Review by oeiyiping — APPROVED** [review]
LGTM.

**Pull request merged**